src/data/fetchFromRmp.ts

```diff
@@ -46,48 +46,49 @@ function getGraphQlUrlProps(professorIds: string[]) {
     return graphQlUrlProps
 }
 
-function fetchWithGraphQl(graphQlUrlProps: any[], resolve) {
+async function fetchWithGraphQl(graphQlUrlProps: any[]) {
     const graphqlUrl = "https://www.ratemyprofessors.com/graphql";
-
-    Promise.all(graphQlUrlProps.map(u=>fetch(graphqlUrl, u)))
-        .then(responses => Promise.all(responses.map(res => res.json())))
-        .then(ratings => {
-            for (let i = 0; i < ratings.length; i++) {
-                if (ratings[i] != null && ratings[i].hasOwnProperty("data") && ratings[i]["data"].hasOwnProperty("node")) {
-                    ratings[i] = ratings[i]["data"]["node"];
-                }
+    let ratings: any[];
```

Comment:
`let ratings: any[];` can be moved to be declared inside the try block `let ratings = await Promise.all(responses.map(res => res.json()));` and then `return ratings` at the end of the try-block. Also add `return null` or `return []` inside the catch-block, and remove `return ratings;` outside of the try-catch.
Also it would be good to make a RMPRatingInterface so that we avoid `any[]`